src/__tests__/lib/containers/oauth/CreateOAuthClient.test.tsx

```diff
@@ -0,0 +1,171 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { createWrapper } from '../../../../lib/testutils/TestingLibraryUtils'
+import { rest, server } from '../../../../mocks/msw/server'
+import {
+  CreateOAuthModal,
+  CreateOAuthModalProps,
+} from '../../../../lib/containers/oauth/CreateOAuthClient'
+import { mockClientList1 } from '../../../../mocks/oauth/MockClient'
+import { displayToast } from '../../../../lib/containers/ToastMessage'
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../../../lib/utils/functions/getEndpoint'
+import { ConfirmModal } from '../../../../lib/containers/oauth/OAuthConfirmationModal'
+
+jest.mock('../../../../lib/containers/ToastMessage', () => {
+  return { displayToast: jest.fn() }
+})
+jest.mock('../../../../lib/containers/oauth/OAuthConfirmationModal', () => ({
+  ConfirmModal: jest.fn().mockImplementation(() => {
+    return <div></div>
+  }),
+}))
+
+const mockConfirmModal = ConfirmModal
+const mockToastFn = displayToast
+
+const updatedClient = jest.fn()
+const mockClient = mockClientList1.results[0]
+const defaultProps: CreateOAuthModalProps = {
+  isShowingModal: true,
+  onClose: jest.fn(),
+  isEdit: false,
+  setSelectedClient: jest.fn(),
+  isShowingConfirmModal: false,
+  setIsShowingConfirmModal: jest.fn(),
+}
+
+function renderComponent(props: CreateOAuthModalProps = defaultProps) {
+  render(<CreateOAuthModal {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+describe('Create 0Auth Client', () => {
+  beforeAll(() => server.listen())
+  afterEach(() => {
+    jest.clearAllMocks()
+    server.restoreHandlers()
+    updatedClient.mockClear()
+  })
+  afterAll(() => server.close())
+
+  it('Shows the modal', async () => {
+    renderComponent()
+    await screen.findByText('Create New OAuth Client')
+    expect(screen.getAllByRole('textbox')).toHaveLength(6)
+  })
+
+  it('Creates a client on save', async () => {
+    server.use(
+      rest.post(
+        `${getEndpoint(
+          BackendDestinationEnum.REPO_ENDPOINT,
+        )}/auth/v1/oauth2/client`,
+        async (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json(req.body))
+        },
+      ),
+    )
+    renderComponent()
+    const inputName = (await screen.findAllByRole('textbox'))[0]
+    const inputHomePage = (await screen.findAllByRole('textbox'))[1]
+    const inputRedirectURI = (await screen.findAllByRole('textbox'))[2]
+
+    const saveButton = screen.queryByRole('button', { name: 'Save' })
```

Comment:
getByRole will throw an error if the button doesn't exist, which seems preferable here.
```suggestion
    const saveButton = screen.getByRole('button', { name: 'Save' })
```